src/Generator/Generators/CSharp/CSharpTextTemplate.cs

```diff
@@ -196,6 +198,111 @@ public void GenerateHeader()
             PopBlock();
         }
 
+        private class DelegateSignature
+        {
+            public Type retType
+            {
+                get;
+                set;
+            }
+            public string retTypeStr
+            {
+                get;
+                set;
+            }
+            public string signature
+            {
+                get;
+                set;
+            }
+            public bool suppressWarning
+            {
+                get;
+                set;
+            }
+            public System.Runtime.InteropServices.CallingConvention callingConvention
+            {
+                get;
+                set;
+            }
+
+            public override int GetHashCode()
+            {
+                return retType.GetHashCode() ^ signature.GetHashCode() ^ callingConvention.GetHashCode();
+            }
+            public override bool Equals(object obj)
+            {
+                return Equals(obj as DelegateSignature);
+            }
+            public bool Equals(DelegateSignature obj)
+            {
+                return obj != null && CompareWith(obj);
+            }
+            private bool CompareWith(DelegateSignature ds)
+            {
+                if (!retType.Equals(ds.retType) || !signature.Equals(ds.signature) || suppressWarning != ds.suppressWarning ||
```

Comment:
This line is always becomes says that the 2 DelegateSignatures are NOT equal as ```retType.Equals(ds.retType)``` always returns false and hence number of delegates increased again! I encountered this once more in past but ignored. Am I missing something here? Is this expected behaviour?!